calico_node/startup/startup.go

```diff
@@ -1014,6 +1014,9 @@ func checkConflictError(err error) error {
 	if conflict, ok := err.(cerrors.ErrorResourceUpdateConflict); ok {
 		log.Infof("Ignoring conflict when setting value %s", conflict.Identifier)
 		return nil
+	} else if exists, ok := err.(cerrors.ErrorResourceAlreadyExists); ok {
```

Comment:
not sure if this is the right place to bundle these two together. Taking a step back, the reason why we do this logic this was:
1. `Get` // to see if the resource is already there (could be a node reboot or a node joining the cluster later)
2. If `Get` didn't return anything then `Create` the resource, so we can set the default values (keep in mind these can change later down the road, so we want to be able to set them to the new default instead of ignoring if the value is already there)
3. If the `Get` in step 1 returned something then we compare it with the defaults we have and if it's not the same then we do an `Update` (this is for the case when we do up update, or change defaults, etc)

So, the question is, are you seeing calico-nodes reboot a lot because of this?

Reply:
Not quite, the settings that are causing issues are global settings, things that every Node tried to set, Copying and modifying your list:

1. Get // to see if the resource is already there (could be a node reboot or a node joining the cluster later)
2. If Get didn't return anything then Create the resource, so we can set the default values (keep in mind these can change later down the road, so we want to be able to set them to the new default instead of ignoring if the value is already there)
3. Another Node sets the global value we previously checked.
4. The Node fails to start and restarts, and now starts because the first Get succeeds

Looking at it again, I might have put this on one or two cases that aren't needed, as local Node configuration shouldn't have this issue, it is only global configurations that we see the race condition pop up.

Reply:
Yup, I was also talking about the global defaults. Ignoring `ResourceAlreadyExists` might have some side effects, corner cases other than the ones I described that I haven't thought of yet. Maybe @caseydavenport or @robbrockbank have some ideas?

Reply:
I'll need to review the code a bit more to be certain of this, but we should only be hitting that error if we call `Create` while another Node has already set it, otherwise we should see a result in the initial `Get` and check/update from there.

Reply:
Personally, I'd opt for removing the `checkConflictError` function altogether, and for each call we make to libcalico handling the expected errors inline there.

That's more readable, and makes it easier to follow this PR.

Reply:
I feel like that might make for a bit of repeated code, but with it catching two errors I think you're right on the readability of it.

I still feel it is worth checking the `ErrorResourceAlreadyExists` on the `Create` calls we do after the `Get` for the global values.

Reply:
It's not that much repeated code, so it's fine.